IconThemePlugin.js

```diff
@@ -0,0 +1,26 @@
+class IconThemePlugin {
+  apply(compiler) {
+    const pluginName = IconThemePlugin.name;
+    const { webpack } = compiler;
+    const { RawSource } = webpack.sources;
+
+    compiler.hooks.emit.tap(pluginName, (compilation) => {
+      const icons = compilation.getAssets().map(
+        (asset) => asset.name.split('/').pop()
+      );
+
+      const iconEntries = icons.map((file) => [
+        file.substr(0, file.lastIndexOf(".")),
+        file.substr(file.lastIndexOf(".") + 1, file.length)
+      ]);
+      const iconTypes = Object.fromEntries(iconEntries);
+
+      const metadata = require('./metadata.json');
```

Comment:
This also relates to what I just reviewed. It is possible that someone has automation to generate stuff. So an option as discussed previously might be a solution for that :thinking: